test/FastTests/Client/Queries/QueryTests.cs

```diff
@@ -32,6 +33,237 @@ private class B
             public byte Byte { get; set; }
         }
 
+        private class Article
+        {
+            public string Title { get; set; }
+            public string Description { get; set; }
+            public bool IsDeleted { get; set; }
+        }
+        
+        [Fact]
+        public async Task Query_CreateClausesForQueryDynamicallyOrOperandAsync()
+        {
+            using (var store = GetDocumentStore())
+            {
+                const string id1 = "users/1";
+                const string id2 = "users/2";
+                
+                using (var session = store.OpenAsyncSession())
+                {
+                   await session.StoreAsync(new Article
+                        {
+                            Title = "foo",
+                            Description = "bar",
+                            IsDeleted = false
+                            
+                        },
+                        id1);
+                    
+                   await session.StoreAsync(new Article
+                        {
+                            Title = "foo",
+                            Description = "bar",
+                            IsDeleted = true
+                        },
+                        id2);
+                   await session.SaveChangesAsync();
+                }
+                
+                using (var session = store.OpenAsyncSession())
+                {
+                    
+                    var query =  session.Advanced.AsyncDocumentQuery<Article>()
+                        .Search(article => article.Title, "foo")
+                        .Search(article => article.Description, "bar", @operator: SearchOperator.Or);
+
+                    var result =  await query.OrElse(wrapPreviousQueryClauses:true).WhereEquals(x => x.IsDeleted, false).ToListAsync();
```

Comment:
This is a problem, that doesn't do what the user likely think it does

Reply:
Could you elaborate?